### ISSUE TITLE
False positive error for datalog or-join syntax.

### ISSUE THREAD

**Issue by rap1ds** [borkdude/clj-kondo#889]
**version**

clj-kondo v2020.05.09

**platform**

Native, docker image and JVM

**problem**

False positive error for datalog or-join syntax.

**repro**

```
clojure -A:clj-kondo --lint - <<< "
(ns group
  (:require [datomic.api :as d]))

(defn can-access-group? [db user-id group-id]
  (boolean
   (d/q '[:find ?user-id .
          :in $ ?user-id ?group-id
          :where [?user :user/id ?user-id]
                 (or-join [?user ?group-id]
                          (and [?user :user/group ?g]
                               [?g :group/id ?group-id])
                          [?user :user/admin true])]
        db user-id group-id)))
"

<stdin>:7:9: error: Join variable not declared inside clauses: [?group-id]
linting took 102ms, errors: 1, warnings: 0
```

**expected behavior**

Datalog syntax is valid so expected output is 0 errors, 0 warnings

**how to fix**

This is already fixed in newer version of [datalog-parser](https://github.com/lambdaforge/datalog-parser/commit/85a3ca1505a4cd250d143086da794328eff59339). I'll make a PR to bump up datalog-parser version.
